File: SepetYorumla.Models/Entities/Review.cs
using SepetYorumla.Core.Entities;

namespace SepetYorumla.Models.Entities;

public class Review : Entity<Guid>
{
  public decimal? StarRating { get; set; }
  public bool? IsThumbsUp { get; set; }
  public bool IsActive { get; set; } = true;

  // Navigation properties
  public Guid UserId { get; set; }
  public virtual User User { get; set; } = default!;
  public Guid BasketId { get; set; }
  public virtual Basket Basket { get; set; } = default!;
}

File: sepetyorumla.client/src/models/Review.ts
export interface ReviewResponseDto {
  id: string;
  starRating: number;
  isThumbsUp: boolean;
  userId: string;
  username: string;
  basketId: string;
  basketTitle: string;
}

export interface UpsertedReviewResponseDto {
  id: string;
  starRating: number;
  isThumbsUp: boolean;
}

export interface UpsertReviewRequest {
  starRating: number | null;
  isThumbsUp: boolean | null;
  basketId: string;
}